enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,199 @@
+type LoadBalancerStrategy struct {
+    // <snip>
+
+    // manageDNS indicates if the lifecyle of the wildcard DNS record
+    // associated with the load balancer service will be managed by
+    // the ingress operator.
+    //
+	// +kubebuilder:default:=true
+	// +kubebuilder:validation:Optional 
+    // +optional
+    ManageDNS bool `json:"manageDNS"`
+}
+```
+
+Re-uses existing state `DNSManaged` of the controller to denote to denote if
+the wildcard *DNSRecord* associated with the controller is managed/un-managed. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `managedDNS`, the controller decides whether to ensure creating
+the a wildcard *DNSRecord*. If the customer is updating `managedDNS` from 
+`"true"` to `"false"` the previously created DNSRecord will be deleted.
+
+The support provided by the installer and this new field aren't entriely
+mutually exclusive i.e. if customer has opted to disable DNS management cluster
+wide (as mentioned under UPI based installations), this additional flag is of
+no value and need not be set since *DNSRecord* created is silently reconciled 
+by the dns controller.
+
+### Risks and Mitigations
+
+The manual creation of the required DNS records will need to be done prior to
+making this change as the canary controller will try to ensure the if the ingress
+domains are reachable, otherwise the `ingress` *ClusterOperator* results in a 
+degraded state. This needs to be documented so as to avoid unecessary support cases.
+
+### Drawbacks
+
+There is an inconsistency in how the existing support is added to disable
+DNS management cluster-wide by the ingress controller i.e. the controller 
+proceedes to create a wildcard *DNSRecord* and during reconcilliation the dns
+controller [silently skips](https://github.com/openshift/cluster-ingress-operator/blob/972f09b9dbb181ae5c414da2a990b57c60fde9d8/pkg/operator/controller/dns/controller.go#L261-L311)
+publishing the record to the zone. Where as in the proposed solution, 
+a wildcard *DNSRecord* is **not** created when `manageDNS` is set to `false`. 
+
+## Design Details
+
+### Test Plan
+
+- Test by updating the default ingresscontroller to be associated with a manually
+  created dns record and ensure correct working of test workload.
+- Test to ensure clean up old DNS records when updating `manageDNS` to false.
+
+
+### Graduation Criteria
+
+N/A.
+
+#### Dev Preview -> Tech Preview
+
+N/A.  This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A.  This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A.
+
+### Upgrade / Downgrade Strategy
+
+On upgrades, this field will default to `true` which is consistent with the
+older versions of the ingress operator.
+
+On downgrades, 
+- If downgrading to an unsupported version while having the `manageDNS` set to
+  `false` could result in failing to create a wildcard *DNSRecord* and therby
+  resulting in the ingress *ClusterOperator* in a degraded state.
```

Comment:
`unsupported version` means the cluster ingress operator which didn't know anything about this field? If so, won't it create non existing `DNSRecord` CRs?

Reply:
Yes, correct. These *DNSRecord* CRs will put the operator in a degraded state. At this point, the only way to recover is to disable DNS management cluster wide. I'll mention the same as well.